sor/src/main/java/com/bazaarvoice/emodb/sor/compactioncontrol/DefaultCompactionControlSource.java

```diff
@@ -28,12 +28,12 @@
     private static final Logger _log = LoggerFactory.getLogger(DefaultCompactionControlSource.class);
 
     private final MapStore<StashRunTimeInfo> _stashStartTimestampInfo;
-    private final Counter _compactionControlTimeCounter;
+    private volatile int _stashStartTimestampMapSize;
 
     @Inject
     public DefaultCompactionControlSource(@StashRunTimeMapStore final MapStore<StashRunTimeInfo> stashStartTimestampInfo, final MetricRegistry metricRegistry) {
         _stashStartTimestampInfo = checkNotNull(stashStartTimestampInfo, "stashStartTimestampInfo");
-        _compactionControlTimeCounter = checkNotNull(metricRegistry, "metricRegistry").counter(MetricRegistry.name("bv.emodb.scan", "CompactionControlSource", "time-count"));
+        checkNotNull(metricRegistry, "metricRegistry").register(MetricRegistry.name("bv.emodb.scan", "CompactionControlSource", "map-size"), (Gauge<Integer>) () -> _stashStartTimestampMapSize);
```

Comment:
The problem I see with this approach is that `_stashStartTimestampMapSize` is a `MapStore` which behind the scenes is implemented using `ZkMapStore`.  This means it's possible for the values to get updated asynchronously due to an ZK update from another node, but this approach will only change the count when the map is updated locally.

A safer approach would be something like `(Gauge<Integer>) () ->  _stashStartTimestampInfo.keySet().size()`.